### ISSUE TITLE
Is RHM differentiable during training?

### ISSUE THREAD

**Issue by willer94** [juhongm999/dhpf#2]
Hi, I note that you used `torch.no_grad()` before applying RHM.
I think the RHM is not differentiable during training. I have three questions here.
1. What's reason cause this problem?
2. And if the RHM is differnetiable, could HPF and DHPF finetune the backbone during training?
3.  In CHM, the 6D conv. module is replace by RHM in the ablation study, is the procedure the sam as [here,](https://github.com/juhongm999/dhpf/blob/master/model/dhpf.py#L60)  using the `geometric_scores` without gradients to re-weight the appearance matching scores?
Many thanks!

**Comment by juhongm999** [comment]
Without `torch.no_grad()`, the code would run smoothly. 
The reason why we stopped gradients in the code block is to save memory during training.
We empirically found the difference between stop-gradient, and differentiable versions is marginal.
The same code block is used in CHM ablation study.

By the way, HPF do not optimize the network parameters.
And, even if RHM is not differentiable, you can still finetune the backbone network in DHPF (computational graph is connected all the way to the backbone network).

**Comment by willer94** [comment]
OK! Thanks!
